ckan/cli/notify.py

```diff
@@ -1,23 +1,40 @@
 # encoding: utf-8
 
+from logging import getLogger
 import click
 from ckan.model import Session, Package, DomainObjectOperation
 from ckan.model.modification import DomainObjectModificationExtension
+from ckan.views.resource import NotAuthorized, ValidationError
 
 
-@click.group(
-    name=u'notify',
-    short_help=u'Send out modification notifications.'
-)
+log = getLogger(__name__)
+
+
+@click.group(name="notify", short_help="Send out modification notifications.")
 def notify():
     pass
 
 
-@notify.command(
-    name=u'replay',
-    short_help=u'Send out modification signals.'
-)
+@notify.command(name="replay", short_help="Send out modification signals.")
 def replay():
     dome = DomainObjectModificationExtension()
     for package in Session.query(Package):
         dome.notify(package, DomainObjectOperation.changed)
+
+
+@notify.command(name="send_emails", short_help="Send out Email notifications.")
+@click.pass_context
+def send_emails(ctx: click.Context):
+    import ckan.logic as logic
+    import ckan.lib.mailer as mailer
+    from ckan.types import Context
+    from typing import cast
+
+    flask_app = ctx.meta["flask_app"]
+    site_user = logic.get_action("get_site_user")({"ignore_auth": True}, {})
+    context = cast(Context, {"user": site_user["name"]})
+    with flask_app.test_request_context():
```

Comment:
@pdelboca  I get: `RuntimeError: Working outside of request context` without that

Reply:
Can you paste the traceback error? We shouldn't need a request context to execute this command. Also, `test_request_context()` is designed only for testing purposes. So in case of needing one we should create it in another way.

Reply:
![image](https://user-images.githubusercontent.com/72216462/157681742-148ac27b-5aa6-4dd9-90f7-e2f5a8947d29.png)

Reply:
and it fails here https://github.com/ckan/ckan/blob/1e86d3ebeff308c99a68de5c0219604337907cd3/ckan/lib/email_notifications.py#L221

Reply:
I see... I have encountered this error before.

The main problem is that we are using `base.render()` to render the email HTML but it doesn't make too much sense to require a Flask request context to render a plain HTML email. (Why do we need to call `_allow_caching()` to render a template?

Can you try to use Flask's plain `render_template` instead of `base.render()`?

Reply:
> I'm not sure if a function like this exists in core somewhere

@amercader does not exist, should I make one?

Reply:
Here is my solution, though, I m not so happy because we are sending  (helpers, site_title, and ungettext).
This could be ok if the CLI command is the only way to access  `send_email_notifications`...

Reply:
> 
> * We shouldn't depend on a request to render an email template from the CLI tool.

We should not, but Flask's standard render_template method disagrees to this
![image](https://user-images.githubusercontent.com/72216462/158033456-9ffa0ffd-d1f6-44a2-a97a-ed613c69e90e.png)
... of course, this can be bypassed with the changes in my latest commit  (you can see it in `remove test_request_context`)  
 but then we are losing `global` variables except the `Config`... If you go to the link you've mentioned (`https://flask.palletsprojects.com/en/2.0.x/templating/#standard-context`) you can confirm that is true as it is clearly stated (`This variable is unavailable if the template was rendered without an active request context.`)

Thanks

Reply:
@TomeCirun @pdelboca @amercader maybe we need to use the [helper]( https://github.com/ckan/ckan/blob/master/ckan/lib/helpers.py#L295)  to get the context in the CLI (as it is stated in the helper comment) .

So instead of calling the [wsgi_app](https://github.com/ckan/ckan/blob/master/ckan/cli/cli.py#L139) we could get the helper to load the context into ctx object

Also there  is a `flask.cli` [decorator](https://flask.palletsprojects.com/en/2.0.x/api/#flask.cli.with_appcontext) which will load the application context but it will require some rework in the whole CLI module

Reply:
@tino097 I think using the `flask.cli` decorator is the way to go for our CLI module since it will provide a lot of benefits. This could be analized and maybe add a new ticket for a refactor.